core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/container/postgres/PostgresConfigService.java

```diff
@@ -60,17 +65,29 @@ public class PostgresConfigService {
     @Inject
     private UpgradeRdsBackupRestoreStateParamsProvider upgradeRdsBackupRestoreStateParamsProvider;
 
+    @Inject
+    private ClusterService clusterService;
+
     public void decorateServicePillarWithPostgresIfNeeded(Map<String, SaltPillarProperties> servicePillar, StackDto stackDto) {
         Map<String, Object> postgresConfig = initPostgresConfig(stackDto);
-
-        RedbeamsDbSslDetails sslDetails = dbCertificateProvider.getRelatedSslCerts(stackDto);
-        Set<String> rootCerts = sslDetails.getSslCerts();
-        if (CollectionUtils.isNotEmpty(rootCerts)) {
-            Map<String, String> rootSslCertsMap = Map.of("ssl_certs", String.join("\n", rootCerts),
-                    "ssl_certs_file_path", dbCertificateProvider.getSslCertsFilePath(),
-                    "ssl_enabled", String.valueOf(sslDetails.isSslEnabledForStack()));
-            servicePillar.put(POSTGRES_COMMON, new SaltPillarProperties("/postgresql/root-certs.sls",
-                    singletonMap("postgres_root_certs", rootSslCertsMap)));
+        String rootCertsBundle;
+        if (certRotationPossible(stackDto)) {
+            boolean sslEnabled;
+            if (StringUtils.isBlank(stackDto.getCluster().getDbSslRootCertBundle())) {
+                RedbeamsDbSslDetails sslDetails = dbCertificateProvider.getRelatedSslCerts(stackDto);
+                rootCertsBundle = clusterService.updateDbSslCert(stackDto.getCluster().getId(), sslDetails);
+                sslEnabled = sslDetails.isSslEnabledForStack();
+            } else {
```

Comment:
I am not sure `StringUtils.isBlank()` is the right condition here. If SSL is disabled, `clusterService.updateDbSslCert()` will set `stackDto.getCluster().dbSslRootCertBundle()` to the empty string, and this property will stay at that value until the cluster gets terminated. In other words, `StringUtils.isBlank()` will always evaluate to `true` in those cases, thus needlessly calling `dbCertificateProvider.getRelatedSslCerts()` (that ultimately relies on redbeams).

I suggest replacing the condition with `stackDto.getCluster().getDbSslRootCertBundle() == null`.